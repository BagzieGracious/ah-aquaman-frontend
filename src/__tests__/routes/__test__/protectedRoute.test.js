import decode from "jwt-decode";
import Authenticate from "../../../routes/protectedRoutes";

describe("protected routes", () => {
  it("should render without crashing", () => {
    expect(Authenticate("token")).toEqual(false);
  });

  it("should render a decoded token", () => {
    const user = {
      username: "Roy",
      email: "roy@andela.com",
      password: "spidermansucks123",
    };
    
    const jwt = require("jsonwebtoken");
    const token = jwt.sign(user, "sfhjjkdfjdkdfjdk", { expiresIn: 60 * 60 * 24 });
    const response = {
      res: jwt.decode(token),
    };
    expect(Authenticate(token)).toEqual(response);
  });
});
