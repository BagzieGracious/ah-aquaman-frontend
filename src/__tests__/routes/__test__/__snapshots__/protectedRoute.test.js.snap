// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`protected routes should protectedRoute component should render without crashing 1`] = `ShallowWrapper {}`;
